pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,324 @@
+import os
+import posixpath
+import shutil
+
+from pyiron_base import GenericJob, DataContainer
+from pyiron_base.generic.filedata import FileData
+from pyiron_contrib.generic.s3io import FileS3IO
+
+
+def _remove_s3_working_directory(file_s3_io_handle, group_to_remove):
+    # Upon moving s3 to base, this should be moved to pyiron_base/job/util.py
+    if file_s3_io_handle.is_dir(group_to_remove):
+        file_s3_io_handle.remove_group(path=group_to_remove)
+
+
+class StorageType(DataContainer):
+    def __init__(self, *, init=None, storage_type=None, table_name=None):
+        """What type of storage is used
+
+        Parameters:
+            storage_type(str): one of ['local', 's3']
+        """
+        super().__init__(init=init, table_name=table_name)
+        self._available_storage_types = ['local', 's3']
+        self._storage_type = None
+        self.storage_type = storage_type
+
+    def create_group(self, name):
+        """
+        Add a new empty subcontainer under the given key.
+
+        Args:
+            name (str): key under which to store the new subcontainer in this container
+
+        Returns:
+            DataContainer: the newly created subcontainer
+
+        >>> pl = DataContainer({})
+        >>> pl.create_group("group_name")
+        DataContainer([])
+        >>> list(pl.group_name)
+        []
+        """
+        self[name] = DataContainer()
+        return self[name]
+
+    @classmethod
+    def _read_only_error(cls):
+        raise RuntimeError("The storage type cannot be changed when it is already in use.")
+
+    @property
+    def storage_type(self):
+        return self._storage_type
+
+    @storage_type.setter
+    def storage_type(self, storage_type):
+        if storage_type not in self._available_storage_types:
+            raise ValueError(f"Expected 'storage_type' to be one of {self._available_storage_types} "
+                             f"but got {storage_type}.")
+        self._storage_type = storage_type
+        for storage in self._available_storage_types:
+            self[storage] = False
+        self[storage_type] = True
```

Comment:
I think this is cool, but it makes me nervous. If I understand correctly 

```python
store = StorageType()
store.storage_type = 'S3'
print(store.S3)
>>> True
```

works, but 

```python
store = StorageType()
store.S3 = True
print(store.storage_type)
>>> None
```

...or? i.e. we allow two ways to read but only one way to set, with no clear warning that setting in the secondary way won't work!

You might be able to get around this by messing with the generic setters and screening for keys in `_available_storage_types`, but I'm not convinced it's worth it.

Reply:
Yes, you are right! That is indeed dangerous...